src/ScssAssetConverter.php

```diff
@@ -66,18 +66,18 @@ public function convert($asset, $basePath)
         return $cssAsset;
     }
 
-    private function getExtension(string $filename): string
+    protected function getExtension(string $filename): string
     {
         return pathinfo($filename, PATHINFO_EXTENSION);
     }
 
-    private function replaceExtension(string $filename, string $newExtension): string
+    protected function replaceExtension(string $filename, string $newExtension): string
     {
         $extensionlessFilename = pathinfo($filename, PATHINFO_FILENAME);
         return "$extensionlessFilename.$newExtension";
     }
 
-    private function convertAndSaveIfNeeded(string $inFile, string $outFile)
+    protected function convertAndSaveIfNeeded(string $inFile, string $outFile)
```

Comment:
This is a bit of a strange one if you ask me. The method logic simply checks if conversion is needed, then does the conversion and finally save this. What's the use case?

Checking if conversion is needed is done is `$this->shouldConvert()` so if it's useful to override this then that method should be made protected instead.

Doing the actual conversion is done in `$this->compiler->compile()` which is already overridable in this PR by providing a custom Compiler implementation.

Saving is done in `$this->storage->put()` which is a thin shell around `file_put_contents()`. If that is undesired (but I wouldn't know why, exactly) then this can be customized by injecting a custom `Storage` implementation or we can move the decision whether to save it to another method and make that overridable. But for that I would first like to know the use case.